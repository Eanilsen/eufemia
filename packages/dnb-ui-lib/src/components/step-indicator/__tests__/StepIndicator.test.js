/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  axeComponent,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../StepIndicator'

// just to make sure we re-run the test in watch mode due to changes in theese files
import _step_indicator from '../style/_step-indicator.scss' // eslint-disable-line
import dnb_step_indicator from '../style/dnb-step-indicator.scss' // eslint-disable-line
import dnb_step_indicator_theme_ui from '../style/themes/dnb-step-indicator-theme-ui.scss' // eslint-disable-line

const props = fakeProps(require.resolve('../StepIndicator'), {
  optional: true
})
delete props.render

const active_url = '?b'
const stepIndicatorListData = [
  {
    title: 'Aa',
    url: '?a'
  },
  {
    title: 'Bb',
    url: '?b'
  },
  {
    title: 'Cc',
    url: '?c',
    url_future: ''
  }
]

describe('StepIndicator component', () => {
  const Comp = mount(
    <Component
      {...props}
      data={stepIndicatorListData}
      active_url={active_url}
    />
  )

  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('have a "active_url" state have to be same as prop from startup', () => {
    expect(Comp.state().active_url).toBe(active_url)
  })

  it('should validate with ARIA rules', async () => {
    expect(
      await axeComponent(Comp, {
        rules: {
          // because of the role="text", we disable this rule for now
          'aria-roles': { enabled: false }
        }
      })
    ).toHaveNoViolations()
  })
})

describe('StepIndicator scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/dnb-step-indicator.scss')
    )
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-step-indicator-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
