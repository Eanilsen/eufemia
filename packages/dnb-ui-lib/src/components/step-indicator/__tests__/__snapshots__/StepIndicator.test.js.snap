// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIndicator component have to match snapshot 1`] = `
<StepIndicator
  active_item="active_item"
  active_url="?b"
  class="class"
  className="className"
  data={
    Array [
      Object {
        "title": "Aa",
        "url": "?a",
      },
      Object {
        "title": "Bb",
        "url": "?b",
      },
      Object {
        "title": "Cc",
        "url": "?c",
        "url_future": "",
      },
    ]
  }
  on_change={[Function]}
  show_numbers="show_numbers"
>
  <div
    className="dnb-step-indicator className class"
  >
    <ul
      className="dnb-step-indicator__list"
    >
      <li
        className="dnb-step-indicator__item dnb-step-indicator--visited"
        key="bc0"
      >
        <ItemContent
          activeItem={1}
          number={0}
          onChangeHandler={[Function]}
          show_numbers="show_numbers"
          title="Aa"
          url="?a"
          url_future={null}
          url_passed={null}
        >
          <a
            className="dnb-anchor dnb-step-indicator__item-content dnb-step-indicator__item-content--link"
            href="?a"
            onClick={[Function]}
          >
            <ItemContentWrapper>
              <span
                className="dnb-step-indicator__item-content--text"
              >
                Aa
                <Dummy>
                  <span
                    aria-hidden={true}
                    hidden={true}
                  >
                    Aa
                  </span>
                </Dummy>
              </span>
            </ItemContentWrapper>
          </a>
        </ItemContent>
      </li>
      <li
        className="dnb-step-indicator__item dnb-step-indicator--active"
        key="bc1"
      >
        <ItemContent
          activeItem={1}
          number={1}
          onChangeHandler={[Function]}
          show_numbers="show_numbers"
          title="Bb"
          url="?b"
          url_future={null}
          url_passed={null}
        >
          <a
            aria-current={true}
            className="dnb-anchor dnb-step-indicator__item-content dnb-step-indicator__item-content--link"
            href="?b"
            onClick={[Function]}
          >
            <ItemContentWrapper>
              <span
                className="dnb-step-indicator__item-content--text"
              >
                Bb
                <Dummy>
                  <span
                    aria-hidden={true}
                    hidden={true}
                  >
                    Bb
                  </span>
                </Dummy>
              </span>
            </ItemContentWrapper>
          </a>
        </ItemContent>
      </li>
      <li
        className="dnb-step-indicator__item"
        key="bc2"
      >
        <ItemContent
          activeItem={1}
          number={2}
          onChangeHandler={[Function]}
          show_numbers="show_numbers"
          title="Cc"
          url="?c"
          url_future=""
          url_passed={null}
        >
          <span
            aria-disabled={true}
            className="dnb-step-indicator__item-content dnb-step-indicator__item-content--static"
            role="text"
          >
            <ItemContentWrapper>
              <span
                className="dnb-step-indicator__item-content--text"
              >
                Cc
                <Dummy>
                  <span
                    aria-hidden={true}
                    hidden={true}
                  >
                    Cc
                  </span>
                </Dummy>
              </span>
            </ItemContentWrapper>
          </span>
        </ItemContent>
      </li>
    </ul>
  </div>
</StepIndicator>
`;

exports[`StepIndicator scss have to match default theme snapshot 1`] = `
"/*
* StepIndicator theme
*
*/
/**
 * This file is only used to make themes independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
.dnb-step-indicator__item-content--static {
  color: var(--color-emerald-green); }

.dnb-step-indicator__item-content--link {
  color: var(--color-sea-green-alt); }

.dnb-step-indicator__item {
  margin-left: 4.5rem; }
  .dnb-step-indicator__item:first-child {
    margin-left: 0; }
  @media screen and (max-width: 50em) {
    .dnb-step-indicator__item {
      margin-left: 2rem; } }

.dnb-step-indicator::after {
  color: var(--color-mint-green); }

.dnb-step-indicator--active.dnb-step-indicator__item::after,
.dnb-step-indicator--active .dnb-step-indicator__item-content::after {
  background-color: var(--color-sea-green-alt); }

.dnb-step-indicator span[hidden],
.dnb-step-indicator--active .dnb-step-indicator__item-content {
  font-weight: var(--font-weight-demi); }
  .dnb-step-indicator span[hidden]:not(:hover),
  .dnb-step-indicator--active .dnb-step-indicator__item-content:not(:hover) {
    color: var(--color-emerald-green); }
"
`;

exports[`StepIndicator scss have to match snapshot 1`] = `
"/*
* DNB StepIndicator
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
.dnb-step-indicator {
  /**
   * 1. Remove repeating backgrounds in all browsers (opinionated).
   * 2. Add border box sizing in all browsers (opinionated).
   */
  /**
   * 1. Add text decoration inheritance in all browsers (opinionated).
   * 2. Add vertical alignment inheritance in all browsers (opinionated).
   */
  /* Grouping content
   */
  /**
   * 1. Add the correct sizing in Firefox.
   * 2. Show the overflow in Edge and IE.
   */
  /**
   * Add the correct display in IE.
   */
  /**
   * Remove the list style on navigation lists in all browsers (opinionated).
   */
  /**
   * 1. Use the default monospace user interface font
   *    in all browsers (opinionated).
   * 2. Correct the odd \`em\` font sizing in all browsers.
   */
  /* Text-level semantics
   */
  /**
   * Remove the gray background on active links in IE 10.
   */
  /**
   * Add the correct text decoration in Edge, IE, Opera, and Safari.
   */
  /**
   * Add the correct font weight in Chrome, Edge, and Safari.
   */
  /**
   * 1. Use the default monospace user interface font
   *    in all browsers (opinionated).
   * 2. Correct the odd \`em\` font sizing in all browsers.
   */
  /*
   * Remove the text shadow on text selections in Firefox 61- (opinionated).
   * 1. Restore the coloring undone by defining the text shadow
   *    in all browsers (opinionated).
   */
  /* Embedded content
   */
  /*
   * Change the alignment on media elements in all browers (opinionated).
   */
  /**
   * Add the correct display in IE 9-.
   */
  /**
   * Add the correct display in iOS 4-7.
   */
  /**
   * Remove the border on images inside links in IE 10-.
   */
  /**
   * Change the fill color to match the text color in all browsers (opinionated).
   */
  /**
   * Hide the overflow in IE.
   */
  /* Tabular data
   */
  /**
   * Collapse border spacing in all browsers (opinionated).
   */
  /* Forms
   */
  /**
   * Inherit styling in all browsers (opinionated).
   */
  /**
   * Remove the margin in Safari.
   */
  /**
   * 1. Show the overflow in IE.
   * 2. Remove the inheritance of text transform in Edge, Firefox, and IE.
   */
  /**
   * Correct the inability to style clickable types in iOS and Safari.
   */
  /**
   * Correct the padding in Firefox.
   */
  /**
   * Show the overflow in Edge and IE.
   * 1. Safari did inherit the break-word from html root
   */
  /**
   * 1. Correct the text wrapping in Edge and IE.
   * 2. Correct the color inheritance from \`fieldset\` elements in IE.
   */
  /**
   * 1. Add the correct display in Edge and IE.
   * 2. Add the correct vertical alignment in Chrome, Firefox, and Opera.
   */
  /**
   * Remove the inheritance of text transform in Firefox.
   */
  /**
   * 2. Remove the default vertical scrollbar in IE.
   * 3. Change the resize direction on textareas in all browsers (opinionated).
   */
  /**
   * Remove the padding in IE 10-.
   */
  /**
   * 1. Correct the odd appearance in Chrome and Safari.
   * 2. Correct the outline style in Safari.
   */
  /**
   * Correct the cursor style of increment and decrement buttons in Safari.
   */
  /**
   * Correct the text style of placeholders in Chrome, Edge, and Safari.
   */
  /**
   * Remove the inner padding in Chrome and Safari on macOS.
   */
  /**
   * 1. Correct the inability to style clickable types in iOS and Safari.
   * 2. Change font properties to \`inherit\` in Safari.
   */
  /**
   * Remove the inner border and padding of focus outlines in Firefox.
   */
  /* Interactive
   */
  /*
   * Add the correct display in Edge and IE.
   */
  /*
   * Add the correct display in all browsers.
   */
  /* Scripting
   */
  /**
   * Add the correct display in IE 9-.
   */
  /**
   * Add the correct display in IE.
   */
  /* User interaction
   */
  /*
   * 1. Remove the tapping delay in IE 10.
   * 2. Remove the tapping delay on clickable elements
        in all browsers (opinionated).
   */
  /**
   * Add the correct display in IE 10-.
   */
  /* Accessibility
   */
  /**
   * Change the cursor on busy elements in all browsers (opinionated).
   */
  /*
   * Change the cursor on control elements in all browsers (opinionated).
   */
  /*
   * Change the cursor on disabled, not-editable, or otherwise
   * inoperable elements in all browsers (opinionated).
   */
  /*
   * Change the display on visually hidden accessible elements
   * in all browsers (opinionated).
   */
  font-family: sans-serif;
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: 1rem;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333);
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\";
  font-size: 1em;
  line-height: 1.5em; }
  .dnb-step-indicator *,
  .dnb-step-indicator ::before,
  .dnb-step-indicator ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-step-indicator ::before,
  .dnb-step-indicator ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }
  .dnb-step-indicator hr {
    height: 0;
    /* 1 */
    overflow: visible;
    /* 2 */ }
  .dnb-step-indicator main {
    display: block; }
  .dnb-step-indicator nav ol,
  .dnb-step-indicator nav ul {
    list-style: none; }
  .dnb-step-indicator pre {
    font-family: Menlo, Consolas, Roboto Mono, Ubuntu Monospace, Noto Mono, Oxygen Mono, Liberation Mono, monospace;
    /* 1 */
    font-size: 1em;
    /* 2 */ }
  .dnb-step-indicator a {
    background-color: transparent; }
  .dnb-step-indicator abbr[title] {
    text-decoration: underline;
    text-decoration: underline dotted; }
  .dnb-step-indicator b,
  .dnb-step-indicator strong {
    font-weight: bolder; }
  .dnb-step-indicator code,
  .dnb-step-indicator kbd,
  .dnb-step-indicator samp {
    font-family: Menlo, Consolas, Roboto Mono, Ubuntu Monospace, Noto Mono, Oxygen Mono, Liberation Mono, monospace;
    /* 1 */
    font-size: 1em;
    /* 2 */ }
  .dnb-step-indicator ::-moz-selection {
    background-color: #b3d4fc;
    /* 1 */
    color: #000;
    /* 1 */
    text-shadow: none; }
  .dnb-step-indicator audio,
  .dnb-step-indicator canvas,
  .dnb-step-indicator iframe,
  .dnb-step-indicator img,
  .dnb-step-indicator svg,
  .dnb-step-indicator video {
    vertical-align: middle; }
  .dnb-step-indicator audio,
  .dnb-step-indicator video {
    display: inline-block; }
  .dnb-step-indicator audio:not([controls]) {
    display: none;
    height: 0; }
  .dnb-step-indicator img {
    border-style: none; }
  .dnb-step-indicator svg:not([fill]) {
    fill: currentColor; }
  .dnb-step-indicator svg:not(:root) {
    overflow: hidden; }
  .dnb-step-indicator table {
    border-collapse: collapse; }
  .dnb-step-indicator button,
  .dnb-step-indicator input,
  .dnb-step-indicator select,
  .dnb-step-indicator textarea {
    font-family: inherit;
    font-size: inherit;
    line-height: inherit; }
  .dnb-step-indicator button,
  .dnb-step-indicator input,
  .dnb-step-indicator select {
    margin: 0; }
  .dnb-step-indicator button {
    overflow: visible;
    /* 1 */
    text-transform: none;
    /* 2 */ }
  .dnb-step-indicator button,
  .dnb-step-indicator [type='button'],
  .dnb-step-indicator [type='reset'],
  .dnb-step-indicator [type='submit'] {
    -webkit-appearance: button; }
  .dnb-step-indicator fieldset {
    padding: 0.35rem 0.75rem 0.625rem; }
  .dnb-step-indicator input {
    overflow: visible;
    word-break: normal;
    /* 1 */ }
  .dnb-step-indicator legend {
    color: inherit;
    /* 2 */
    display: table;
    /* 1 */
    max-width: 100%;
    /* 1 */
    white-space: normal;
    /* 1 */ }
  .dnb-step-indicator progress {
    display: inline-block;
    /* 1 */
    vertical-align: baseline;
    /* 2 */ }
  .dnb-step-indicator select {
    text-transform: none; }
  .dnb-step-indicator textarea {
    overflow: auto;
    /* 2 */
    resize: vertical;
    /* 3 */ }
  .dnb-step-indicator [type='checkbox'],
  .dnb-step-indicator [type='radio'] {
    padding: 0; }
  .dnb-step-indicator [type='search'] {
    -webkit-appearance: textfield;
    /* 1 */
    outline-offset: -2px;
    /* 2 */ }
  .dnb-step-indicator ::-webkit-inner-spin-button,
  .dnb-step-indicator ::-webkit-outer-spin-button {
    height: auto; }
  .dnb-step-indicator ::-webkit-input-placeholder {
    color: inherit;
    opacity: 0.54; }
  .dnb-step-indicator ::-webkit-search-decoration {
    -webkit-appearance: none; }
  .dnb-step-indicator ::-webkit-file-upload-button {
    -webkit-appearance: button;
    /* 1 */
    font: inherit;
    /* 2 */ }
  .dnb-step-indicator ::-moz-focus-inner {
    border-style: none;
    padding: 0; }
  .dnb-step-indicator details {
    display: block; }
  .dnb-step-indicator summary {
    display: list-item; }
  .dnb-step-indicator canvas {
    display: inline-block; }
  .dnb-step-indicator template {
    display: none; }
  .dnb-step-indicator a,
  .dnb-step-indicator area,
  .dnb-step-indicator button,
  .dnb-step-indicator input,
  .dnb-step-indicator label,
  .dnb-step-indicator select,
  .dnb-step-indicator summary,
  .dnb-step-indicator textarea,
  .dnb-step-indicator [tabindex] {
    -ms-touch-action: manipulation;
    /* 1 */
    touch-action: manipulation;
    /* 2 */ }
  .dnb-step-indicator [hidden] {
    display: none; }
  .dnb-step-indicator [aria-busy='true'] {
    cursor: progress; }
  .dnb-step-indicator [aria-controls] {
    cursor: pointer; }
  .dnb-step-indicator [aria-disabled='true'],
  .dnb-step-indicator [disabled] {
    cursor: not-allowed; }
  .dnb-step-indicator [aria-hidden='false'][hidden]:not(:focus) {
    clip: rect(0, 0, 0, 0);
    display: inherit;
    position: absolute; }
  @media print {
    .dnb-step-indicator {
      background-color: white; } }

/*
 * StepIndicator component
 *
 */
.dnb-step-indicator {
  position: relative;
  z-index: 1;
  display: block;
  margin: 0;
  padding: 0; }
  .dnb-step-indicator__list {
    display: flex;
    flex-direction: row;
    align-items: stretch;
    padding: 0;
    margin: 0;
    list-style: none; }
  .dnb-spacing .dnb-step-indicator__list,
  .dnb-spacing .dnb-step-indicator__list li {
    margin-top: 0;
    margin-bottom: 0; }
  .dnb-step-indicator__item-contenta, .dnb-step-indicator__item-content.dnb-anchor, .dnb-step-indicator__item-content--static, .dnb-step-indicator__item-content--link {
    position: relative;
    z-index: 2;
    display: flex;
    flex-direction: row;
    margin: 0;
    padding: 0.25rem 0.25rem;
    height: 100%;
    border-bottom: none;
    text-decoration: none;
    box-shadow: none; }
  :not([dnb-is-touch]) .dnb-step-indicator__item-content--link:focus[disabled] {
    cursor: not-allowed; }
  :not([dnb-is-touch]) .dnb-step-indicator__item-content--link:focus:not([disabled]) {
    outline: none; }
    html[data-whatinput='keyboard'] :not([dnb-is-touch]) .dnb-step-indicator__item-content--link:focus:not([disabled]) {
      box-shadow: 0 0 0 0.125rem var(--color-emerald-green);
      border-color: transparent; }
    :not([dnb-is-touch]) .dnb-step-indicator__item-content--link:focus:not([disabled])::after {
      content: none; }
  .dnb-step-indicator__item-content--number {
    padding-right: 0.25rem;
    white-space: nowrap; }
  .dnb-step-indicator__item-content--text > span[hidden] {
    display: flex;
    flex-direction: column;
    height: 0;
    visibility: hidden;
    overflow: hidden; }
  .dnb-step-indicator__item-content::after {
    content: '';
    position: absolute;
    z-index: 1;
    left: 0;
    bottom: 0;
    width: 100%;
    height: 3px;
    border-radius: 1.5px; }
  .dnb-step-indicator::after {
    content: '';
    position: absolute;
    z-index: -1;
    left: -50vw;
    bottom: 0;
    width: 100vw;
    height: 1px;
    color: black;
    background-color: currentColor;
    box-shadow: 100vw 0 0 0 currentColor; }
"
`;
